### PR TITLE
Compatibility with API 1.0.0

### PR DESCRIPTION
**Pull request by Kapim**

### CODE REVIEW

File: arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs [hunk 1]
```diff
@@ -270,6 +273,24 @@ public class Parameter : IO.Swagger.Model.ActionParameter {
...
+
+        public static IParameter InitializePositionParameter(ParameterMetadata actionParameterMetadata, VerticalLayoutGroup layoutGroupToBeDisabled, GameObject canvasRoot, OnChangeParameterHandlerDelegate onChangeParameterHandler, string value, string type, bool linkable, CanvasGroup windowToHideWhenRequestingObj, ActionPoint actionPoint) {
+
+            LinkableDropdownPositions dropdownParameter = GameObject.Instantiate(ActionsManager.Instance.LinkableParameterDropdownPositionsPrefab).GetComponent<LinkableDropdownPositions>();
+            string selectedValue = null;
+            if (value != null) {
+                selectedValue = Parameter.GetValue<string>(value);
+            }
+            dropdownParameter.Init(actionParameterMetadata, type, selectedValue, layoutGroupToBeDisabled, canvasRoot, onChangeParameterHandler, windowToHideWhenRequestingObj, actionPoint, linkable);
+            //DropdownParameterPutData(dropdownParameter.DropdownParameter, data, selectedValue, actionParameterMetadata.Name, onChangeParameterHandler, actionParameterMetadata.Type);
```
Comment (ZdenekM) [inline arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs:286]: Commented code could be removed. If it is there for some reason, I would add a comment explaining why it is commented out.

File: arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs [hunk 2]
```diff
@@ -609,7 +633,10 @@ public class Parameter : IO.Swagger.Model.ActionParameter {
                     parameter = InitializeRelativePoseParameter(actionParameterMetadata, handler, Parameter.GetValue<IO.Swagger.Model.Pose>(value), linkable);
                     break;
                 case "pose":
-                    parameter = InitializePoseParameter(actionParameterMetadata, layoutGroupToBeDisabled, canvasRoot, handler, value, type, linkable, windowToHideWhenRequestingObj);
+                    parameter = InitializePoseParameter(actionParameterMetadata, layoutGroupToBeDisabled, canvasRoot, handler, value, type, linkable, windowToHideWhenRequestingObj, actionPoint);
+                    break;
+                case "position":
```
Comment (ZdenekM) [inline arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs:638]: Why not use `case POSITION:` as in `ParameterMetadata.cs`?

File: arcor2_AREditor/Assets/TABLET/Scripts/GUI/ActionParametersMenu.cs [hunk 3]
```diff
@@ -65,6 +65,9 @@ public class ActionParametersMenu : RightMenu<ActionParametersMenu>
         if (isValueValid && currentAction.Parameters.TryGetValue(parameterId, out Parameter actionParameter)) {           
             try {
                 if (JsonConvert.SerializeObject(newValue) != actionParameter.Value) {
+                    if (newValue == null) {
+                        actionParameter.SetValue(actionParameter.ParameterMetadata.GetDefaultValue());
```
Comment (ZdenekM) [inline arcor2_AREditor/Assets/TABLET/Scripts/GUI/ActionParametersMenu.cs:69]: Just a tip for simpler code - can be rewritten as `newValue ??= actionParameter...`

File: arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs [hunk 4]
```diff
@@ -14,42 +14,62 @@ public class LinkableDropdownPoses : LinkableDropdown
...
 
     public void Init(ParameterMetadata parameterMetadata, string type, object value, VerticalLayoutGroup layoutGroupToBeDisabled, GameObject canvasRoot,
-        Parameter.OnChangeParameterHandlerDelegate onChangeParameterHandler, CanvasGroup windowToHideWhenRequestingObj, bool linkable = true) {
+        Parameter.OnChangeParameterHandlerDelegate onChangeParameterHandler, CanvasGroup windowToHideWhenRequestingObj, ActionPoint actionPoint, bool linkable = true) {
         base.Init(parameterMetadata, type, value, layoutGroupToBeDisabled, canvasRoot, onChangeParameterHandler, linkable);
         canvasGroupToHide = windowToHideWhenRequestingObj;
         List<string> options = new List<string>();
+        parentActionPoint = actionPoint;
 
         /*foreach (Base.ActionPoint ap in Base.ProjectManager.Instance.GetAllActionPoints()) {
```
Comment (ZdenekM) [inline arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs:27]: Remove?

File: arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs [hunk 5]
```diff
@@ -14,42 +14,62 @@ public class LinkableDropdownPoses : LinkableDropdown
...
             } catch (KeyNotFoundException ex) {
                 Debug.LogError(ex);
             }
 
+        } else {
+
         }
-        if (type == "link")
-            ActionsDropdown.SetValue($"{selectedAP.GetName()}.{selectedOrientation.GetName()}");
+        //if (type == "link")
```
Comment (ZdenekM) [inline arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs:57]: Remove?

**Review by ZdenekM — COMMENTED** [review]
I just spotted a few details...

**Review by ZdenekM — APPROVED** [review]

File: arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs (hunk 4 above)
Comment (ZdenekM) [reply arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs:27]: I meant the commented code on line 27...

File: arcor2_AREditor/Assets/TABLET/Scripts/GUI/ActionParametersMenu.cs (hunk 3 above)
Comment (Kapim) [reply arcor2_AREditor/Assets/TABLET/Scripts/GUI/ActionParametersMenu.cs:69]: Not in this case as the value is not assigned to the "newValue" variable.

Comment (ZdenekM) [reply arcor2_AREditor/Assets/TABLET/Scripts/GUI/ActionParametersMenu.cs:69]: I see, sorry :)

File: arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs (hunk 1 above)
Comment (Kapim) [reply arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs]: done

File: arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs (hunk 2 above)
Comment (Kapim) [reply arcor2_AREditor/Assets/BASE/Scripts/Parameter.cs]: good point, changed at other places as well

File: arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs (hunk 5 above)
Comment (Kapim) [reply arcor2_AREditor/Assets/TABLET/Scripts/GUI/LinkableDropdownPoses.cs]: done

**Pull request merged**